From 5259dc45e2c97e5ff9da99fbebfb6008de30fe95 Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Wed, 19 May 2021 10:19:19 +0530
Subject: [PATCH 1537/1921] coresight: tmc: Fix a trivial uninitialized pointer
 access

Fix a trivial uninitialized pointer access which gets
invoked when secure trace buffer is not available.

Change-Id: I650275b6cc9d4569de1bc3dc9871aa317a685124
Signed-off-by: Linu Cherian <lcherian@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52103
Tested-by: Devapraba Muthumani <dmuthumani@marvell.com>
Reviewed-by: Devapraba Muthumani <dmuthumani@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/hwtracing/coresight/coresight-tmc.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/hwtracing/coresight/coresight-tmc.c b/drivers/hwtracing/coresight/coresight-tmc.c
index 8aa60767fb4f..6b25bd5a2ac6 100644
--- a/drivers/hwtracing/coresight/coresight-tmc.c
+++ b/drivers/hwtracing/coresight/coresight-tmc.c
@@ -541,8 +541,7 @@ static int tmc_probe(struct amba_device *adev, const struct amba_id *id)
 		if (drvdata->etr_options & CSETR_QUIRK_SECURE_BUFF) {
 			if (tmc_get_cpu_tracebufsize(drvdata, &drvdata->size) ||
 			    !drvdata->size) {
-				dev_info(&drvdata->csdev->dev,
-					 "Secure tracebuffer not available\n");
+				pr_err("Secure tracebuffer not available\n");
 				ret = -ENOMEM;
 				goto out;
 			}
-- 
2.31.1

